infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationService.kt

```diff
@@ -729,42 +729,22 @@ class PublicationService @Autowired constructor(
             }
     }
 
-    private fun validateDuplicateOf(
+    @Transactional(readOnly = true)
+    fun validateDuplicateOf(
         version: ValidationVersion<LocationTrack>,
-        locationTrack: LocationTrack,
+        track: LocationTrack,
         validationVersions: ValidationVersions,
     ): List<PublishValidationError> {
+        // Find the versions in validation set context. These can be null (draft-only that's not included)
         val duplicatesAfterPublication = locationTrackDao
-            .fetchDuplicates(version.officialId)
-            .map(locationTrackDao::fetch)
-            .mapNotNull { duplicate -> fetchPostPublicationVersion(validationVersions, duplicate) }
+            .fetchDuplicateIdsInAnyContext(version.officialId)
+            .mapNotNull { id -> getLocationTrack(id, validationVersions) }
             .filter { potentialDuplicate -> potentialDuplicate.duplicateOf == version.officialId }
 
-        val duplicateOf = locationTrack.duplicateOf?.let { duplicateId ->
-            getLocationTrack(duplicateId, validationVersions)
-        }
-        return validateDuplicateOfState(
-            locationTrack,
-            duplicateOf,
-            validationVersions.locationTracks.map { it.officialId },
-            duplicatesAfterPublication,
-        )
-    }
-
-    private fun fetchPostPublicationVersion(
-        validationVersions: ValidationVersions,
-        track: LocationTrack,
-    ): LocationTrack? {
-        val trackIsInPublicationUnit =
-            validationVersions.locationTracks.any { validationVersion -> validationVersion.officialId == track.id }
-
-        return if (trackIsInPublicationUnit && track.isOfficial()) {
-            locationTrackDao.fetchVersion(track.id as IntId, DRAFT)?.let(locationTrackDao::fetch) ?: track
-        } else if (!trackIsInPublicationUnit && track.isEditedDraft()) {
-            locationTrackDao.fetchVersion(track.id as IntId, OFFICIAL)?.let(locationTrackDao::fetch)
-        } else {
-            track
-        }
+        val duplicateOf = track.duplicateOf?.let { id -> getLocationTrack(id, validationVersions) }
+        // Draft-only won't be found if it's not in the publication set -> get name from draft for validation errors
+        val duplicateOfDraftName = track.duplicateOf?.let { id -> locationTrackDao.getOrThrow(DRAFT, id).name }
```

Comment:
Koska itse raidetta ei validointijoukon kontekstissa ole, haetaan virheilmoitusta varten nimi draft-maailmasta piittaamatta kontekstista.

Reply:
Voidaankos näistä tapahtuvia virheitä näyttää koskaan muualla kuin DRAFT-jutuille? Julkaisuvalidaation virheethän näytetään myös OFFICIAL-puolen käsitteille eheys-infobokseissa. Jos näitä virheitä voi koskaan esiintyä siellä (esim. sen takia että Ratkon lähtödatassa on aikanaan ollut jotain outoa eikä sitä ole vielä korjattu,) lienisi oikeampaa parametrisoida nimen hakeminen publishTypen mukaan. Mutta mikäli näitä ei voi koskaan esiintyä tuollaisissa tilanteissa niin sitten ei niin väliä.

Reply:
Ei tätä voi official maailmassa tapahtua, eikä tuota voi sitoa millään publishtypellä. Julkaisujoukko on ainoa asia mitä tänne tulee parametrina eikä voikaan tulla mitään muuta. 

Ratkaistava ongelma on siis se että validoinnissa on aina katsottava julkaisujoukon muodostamaa leikkausta draft-maailmaa, liittettynä officialin päällä. Eli muodostetaan combo jossa on official + osa(!) draft-asioista ja validoidaan että se on ehjä kokonaisuus. Tässä on kyse tilanteesta jossa: 
- Raide X on merkattu duplikaatiksi raiteelle Y ja nyt raidetta X ollaan julkaisemassa
- Jos Y on mukana julkaisujoukossa, validoidaan draft X+draft Y comboa normaalisti (tämä duplikaatinimihaku on turha)
- Jos Y on olemassa officialissa mutta ei julkaisujoukossa, validoidaan combo draft X + official Y (jälleen tämä nimihaku on turha)
- Jos Y on draft-only eikä julkaisujoukossa mukana, validoidaan combo draft X + (olematon! official) Y. Tämä on selvästi epävalidi ja tilanne tunnistetaan validointifunkkarissa, mutta olemattomasta raiteesta on vaikea poimia nimeä virheviestiin. Siksi tässä haetaan nimi draft-puolen oliosta ja se on ainoa paikka mistä se voidaan hakea kun officialia ei kerran ole.